EESSI roadmap 2026 for approval by SC (#680)
EESSI/docs: docs/roadmap.md

> # Roadmap for EESSI

Reviewer: Maybe this should say "Short-term roadmap for EESSI"?

*(approved by EESSI steering committee: dd-mm-yyyy; valid until: dd-mm-yyyy; to be revised by: dd-mm-yyyy; questions/suggestions: contact-sc@eessi.io)*

The purpose of this roadmap is to focus our collective efforts on the specific goals we aim to achieve together this year.

## Core Infrastructure, Operations & Policy
### Lifecycle & Governance Policies

> - Generation Lifecycle: Define clear policies for "Active" vs. "Archived" generations (e.g., how long to add software to EESSI/2023.06)

Reviewer: If we use "generation" here, we should clarify somewhere (not here) what we mean.
Up until now we've commonly used "version", we should stick to that here?

- Bot Modernisation: Upgrade infrastructure; implement tarball analysis to simplify ingestion and track differences
- Maintainer Support: Leverage the EESSI bot to assist EasyBuild maintainers
### Quality Assurance & Compliance
- Automated License Checks: Towards automatic mandatory license checks for EESSI/2026.06
- Monitoring: Establish regular use of regression tests & status dashboard
- Performance: Assess performance of end-user applications
### Compatibility Layer
- Release EESSI/2026.06: next generation bundled with upcoming toolchains (update of Gentoo Prefix)
## Software Stack & Hardware Targets
### Hardware Enablement
- NVIDIA: Support for Blackwell GPUs
- AMD: Support for Zen5 and ROCm stack (AMD GPUs)
- ARM: Support for AWS Graviton 4+5
### Software Portfolio
- AI/ML Focus: Extend GPU software (PyTorch, TensorFlow, AlphaFold)
- Volume Goal: Reach 1,000 unique software packages
- Toolchains: Integration of lfoss/2025b (EESSI/2025.06) and foss/2026* (EESSI/2026.06)
## User Experience & Integration
### Direct User Interaction
- CLI: Prototype an EESSI Command Line Interface (CLI) to provide an interface beyondmodules

> - Discovery: Create a dynamic software overview

Reviewer 2: It doesn't, but the current description is in my view to abstract. What does that mean, a 'dynamic software overview'? What's not dynamic about it currently? Maybe a question for @boegel : why was the previous software overview not good enough? The answer to that should be what's here in the roadmap (e.g. restructure to software overview to accomodate the increasing number of CPU/GPU architectures, implement capabilities to search for architecture support - I don't know what)

### Feedback & Transparency
- Software Wishlist: Implement mechanism (e.g., anonymous forms) for users to request software and trigger documentation PRs

> - Work-in-Progress (WIP) View: Create dashboard/overview of WIP PRs, so users can see upcoming software

Reviewer: We should add the "release" concept here: monthly "releases" (needs better name, maybe "revision") with a changelog (and associated DOI)

Reviewer 3: This sounds cumbersome, why would people use that DOI? If it is automated that's ok.

# Short-term roadmap for EESSI

> *(approved by EESSI steering committee: 19-02-2026; valid until: 18-02-2027; to be revised by: 18-12-2026; questions/suggestions: support@eessi.io)*

Reviewer 2: We should update these dates...

```suggestion
*(approved by EESSI steering committee: 12-05-2026; valid until: 11-05-2027; last update: 12-05-2026; questions/suggestions: support@eessi.io)*
```

I'm actually not sure why the revision date is earlier than the 'valid until' date, because if you mean to revise it in, say, 6 months from now, you can't reallly say it'll be valid for 1 year (you already know it's only going to be valid for 6 months). I'd personally just scrap the 'revised by' date. We can (and probably should) always revise it.

I replaced it with 'last update' so that it's clear that this is a living document, and when it was last updated. That clearly signals that the basis of the document is fixed - and valid for 1 year - but we may do small modifications.

The purpose of this roadmap is to focus the collective effort of the EESSI
community on the specific goals we aim to achieve together in the coming year.

## Core Infrastructure, Operations & Policy
### Lifecycle & Governance Policies
- Version Lifecycle: Define clear policies for "Active" vs. "Archived" versions (e.g., how long to add software to EESSI/2023.06)
- Development Policy: Establish lifetime policies for experimental installations on `dev.eessi.io`

> - Cadence: Formalise the yearly release cycle (targeting pre-summer releases)

Reviewer 2: Is the goal on the roadmap to _define_ the policies, or to also actually implement them?

### Build System Modernisation
- Diversify Build Sites: Enhance reliability by spreading build operations

> - Bot Modernisation: Upgrade infrastructure; implement tarball analysis to simplify ingestion and track differences

Reviewer 2: ```suggestion
- Bot Modernisation: Upgrade infrastructure; implement tarball analysis to facilitate evaluating builds with large numbers of tarballs
```

> - Maintainer Support: Leverage the EESSI bot to assist EasyBuild maintainers

Reviewer 2: I prefer the 2nd. The first doesn't make clear that this bot actually tests integration with EESSI. Actually, I'd prefer a 3rd option:

- ensuring integration of new EasyBuild contributions with EESSI, by performing test builds on top of EESSI in EasyBuild

### Quality Assurance & Compliance
- Automated License Checks: Towards automatic mandatory license checks for the next EESSI version (2026.x)
- Monitoring: Establish regular use of regression tests & status dashboard
- Performance: Assess performance of end-user applications
### Compatibility Layer
- Release the next EESSI version (2026.x): next version bundled with upcoming toolchains (update of Gentoo Prefix)
## Software Stack & Hardware Targets
### Hardware Enablement

> - NVIDIA: Support for Blackwell GPUs

Reviewer 2: We can take this out - it's done.

> - AMD: Support for Zen5 and ROCm stack (AMD GPUs)

Reviewer 2: Zen5 support is _so_ close that I'm not sure if it should still be part of the roadmap - but no big deal if it is.

- ARM: Support for AWS Graviton 4+5
### Software Portfolio
- AI/ML Focus: Extend GPU software (PyTorch, TensorFlow, AlphaFold)
- Volume Goal: Reach 1,000 unique software packages
- Toolchains: Integration of `lfoss/2025b` (in EESSI 2025.06) and `foss/2026*` (in EESSI 2026.x) toolchains
## User Experience & Integration
### Direct User Interaction
- CLI: Prototype an EESSI Command Line Interface (CLI) to provide an interface beyondmodules
- Discovery: Create a dynamic software overview
### Feedback & Transparency

> - Software Wishlist: Implement mechanism (e.g., anonymous forms) for users to request software and trigger documentation PRs

Reviewer: An easy way to implement this would be to open issue in `EESSI/software-layer` repo, along with a bunch of labels to clarifiy context (like `EuroHPC`), priority, etc.

Some kind of voting mechanism on top should be feasible (thumbs up in PR description)

- Work-in-Progress (WIP) View: Create dashboard/overview of WIP PRs, so users can see upcoming software
### Workflow Integration

> - Tools: Prototype Nextflow & Spack integration

Reviewer 2: We did _some_ prototyping of Spack integration - at least by building on top. Does that cover what we planned? If so: should this be removed? If not: should we make more specific what we mean by 'integration'?

> - CI/CD: Promote EESSI usage in CI/CD pipelines

Reviewer 2: Are we planning concrete activities _beyond_ what we already have? I.e. the GH/Gitlab actions are already there - and have been for a long time.

Reviewer: Blog posts, demos in talks, etc.

as priorities evolve and goals that were set have been accomplished.

Most EESSI development currently happens within the scope of funded projects, but many of the goals outlined here benefit
from broader community involvement and collaboration.

Note that these goals are aspirational, we can *not* provide any guarantees that they will indeed be accomplished
in the near future.

In case of questions related to this roadmap, or if you believe important topics are missing from the roadmap,
please reach out to the [EESSI Steering Committee](governance/steering_committee.md), or open a [support ticket](support.md).

## Core Infrastructure

We will continue improving the reliability and scalability of the EESSI build and deployment infrastructure.

Particular attention will be given to distributing operational responsibilities across sites and contributors,
improving build and testing workflows, and strengthening integration with [EasyBuild](https://easybuild.io).

## Hardware Enablement

We will continue expanding support in EESSI for additional hardware platforms,
while growing the collection of available GPU-enabled scientific and AI/ML software.

> Current areas of interest include support newer generations of [AWS Graviton](https://aws.github.io/graviton/) CPUs (4 and 5),

Reviewer 4: ```suggestion
Current areas of interest include support for newer generations of [AWS Graviton](https://aws.github.io/graviton/) CPUs (4 and 5),
```

and support for AMD GPUs through the [ROCm software ecosystem](https://www.amd.com/en/products/software/rocm.html).

## Software Stack

EESSI will continue improving software quality, reproducibility, and transparency across the software stack, with particular attention to software licensing and compliance.

Current areas of interest include automated license compliance workflows and improved visibility of software licensing information.

> In additional, we expect to provide installations for over 1,000 unique (non-extension) software projects soon --

Reviewer 4: ```suggestion
In addition, we expect to provide installations for over 1,000 unique (non-extension) software projects soon --
```

> In addition, we expect to provide installations for over 1,000 unique (non-extension) software projects soon --

Reviewer: ```suggestion
We intend to create a new EESSI version every (calendar) year, by rebuilding the EESSI compatibility layer and gradually installing software on top of it in the software layer, using new compiler toolchains.

In addition, we expect to provide installations for over 1,000 unique (non-extension) software projects soon --
```

@bedroge Let's add this too before merging?
